source/extensions/filters/listener/tls_inspector/tls_inspector.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "envoy/event/file_event.h"
+#include "envoy/event/timer.h"
+#include "envoy/network/filter.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "common/common/logger.h"
+
+#include "openssl/bytestring.h"
+#include "openssl/ssl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace TlsInspector {
+
+/**
+ * All stats for the TLS inspector. @see stats_macros.h
+ */
+#define ALL_TLS_INSPECTOR_STATS(COUNTER)                                                           \
+  COUNTER(connection_closed)                                                                       \
+  COUNTER(client_hello_too_big)                                                                    \
+  COUNTER(read_error)                                                                              \
+  COUNTER(read_timeout)                                                                            \
+  COUNTER(no_tls_found)                                                                            \
+  COUNTER(tls_found)                                                                               \
+  COUNTER(no_sni_found)                                                                            \
+  COUNTER(sni_found)
```

Comment:
This is me being pedantic, but after this is sorted alphabetically, the stats are weirdly intermixed:
```
$ curl -s localhost:9901/stats | grep tls_inspector
tls_inspector.client_hello_too_big: 0
tls_inspector.connection_closed: 0
tls_inspector.no_sni_found: 0
tls_inspector.no_tls_found: 0
tls_inspector.read_error: 0
tls_inspector.read_timeout: 0
tls_inspector.sni_found: 1
tls_inspector.tls_found: 1
```

Could you rename `no_tls_found` to `tls_not_found` and `no_sni_found` to `sni_not_found`, so that they're grouped together? i.e.

```
piotrsikora@piotrsikora:~/Google/envoy$ curl -s localhost:9901/stats | grep tls_inspector
tls_inspector.client_hello_too_big: 0
tls_inspector.connection_closed: 0
tls_inspector.read_error: 0
tls_inspector.read_timeout: 0
tls_inspector.sni_found: 1
tls_inspector.sni_not_found: 0
tls_inspector.tls_found: 1
tls_inspector.tls_not_found: 0
```

Also, nitpicking, but could you rename `client_hello_too_big` to `client_hello_too_large`.